tutorials/arvo/ames.md

```diff
@@ -112,40 +115,299 @@ ducts as unique keys.  Stop tracking a peer by sending Ames a
+because only complete packets ever make it to Ames, as malformed
+packets will be discarded by the Ames I/O module in the
+[King](#the-serf-and-the-king). Therefore there is no need for Ames to do
+something like nack a packet whose checksum does not match the `+mug` of the
+body.
+
+A nack will be accompanied by a naxplanation, which is a third type of message
+separate from `%plea`s and `%boon`s. Ames won't give the vane that requested the
+initial `%plea` a nack until it also receives the naxplanation, which it will send to
+the vane as a `%done` gift. Naxplanations may only be acked, never nacked.
+
+#### (N)ack packets
+
+The contents of a given (n)ack packet are deterministic. The `content` (i.e. the
+encrypted portion of the packet) of a fragment ack is obtained by encrypting the `+jam` of the
+following noun:
+
+```hoon
+[our-life her-life bone message-num fragment-num]
+```
+Each datum in this noun is an atom with the aura `@ud` or an aura that nests
+under `@ud`.
+
+Here, `our-life` refers to the [`life`](@/docs/glossary/breach), or revision
+number, of the acking ship's networking keys, and `her-life` is the `life` of
+the ack-receiving ship's networking keys. `bone` is an opaque number identifying the flow, `message-num` denotes the number of the
+message in the flow identified by `bone`, and
+`fragment-num` denotes the number of the fragment of the message identified by
+`message-num` being acked. 
+
+A message (n)ack is an extended fragment ack that is obtained by encrypting the
+`+jam` of the following noun:
+
+```hoon
+[our-life her-life bone message-num fragment-num ok=? lag=@dr]
+```
+`ok` is a flag that is set to `%.y` for a message ack and `%.n` for a message
+nack. In the future, `lag` will be used to
+describe the time it took to process a message, but for now it is always zero.
+
+
+### Flows
+
+Flows are asymmetric communication channels along which two ships send and
+receive packets, and all Ames packets are part of some flow.
+
+Flows are asymmetric in that one ship is always the sender of
+`%plea`s and the other is always the sender of `%boon`s on a given flow. Thus ships will
+typically have multiple flows between them.
+
+The sequence of `%plea`s in a flow is totally ordered, though
+the packets which make them up need not be. `%boon`s are totally ordered in the
+same fashion, but there is not a coordinated ordering between `%plea`s and
+`%boon`s in a given flow beyond the implicit one arising from the fact that a
+`%boon` is always a response to a `%plea`.
+
+Inside of Ames, each flow has four sequential opaque `@ud`s
+called bones that are unique to that flow. Thus the flow itself is often
+referred to by its first bone. The set of types of packets is partitioned into
+four subsets, each of which is assigned a bone. Each bone is a one-way street
+for packets to travel along, so e.g. acks to packets making up a `%plea` are
+sent along different bones.
```

Comment:
They're not sent on different bones from each other; all acks to `%plea`s on a given flow will be on the same bone.